bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java

```diff
@@ -3083,9 +3119,25 @@ private int getZoomLevelForMapping(Control from, Control to) {
 public Rectangle map (Control from, Control to, Rectangle rectangle) {
 	checkDevice ();
 	if (rectangle == null) error (SWT.ERROR_NULL_ARGUMENT);
-	int zoom = getZoomLevelForMapping(from, to);
-	rectangle = DPIUtil.scaleUp(rectangle, zoom);
-	return DPIUtil.scaleDown(mapInPixels(from, to, rectangle), zoom);
+	return map(from, to, rectangle.x, rectangle.y, rectangle.width, rectangle.height);
+}
+
+private Rectangle getRectangleInPixelsInDisplayCoordinateSystem(int x, int y, int width, int height) {
```

Comment:
Wouldn't it make sense to group them up with the getLocation... methods added above?

Reply:
Done